### PR TITLE
Autosteer key for pointing the ship at the selected ship, asteroid or planet.

### PR DESCRIPTION
**Pull request by ChamEV**
Adds a keyboard key for auto steering your ship to point it at the targeted ship, asteroid, or selected planet giving the player more control of their movement separate from firing.

Named autosteer to avoid confusion with the more advanced autopilot functions.  Properly breaks autopilot jump/landing/boarding as other input controls do. 

Rearranges the preference panel and puts cloak in it's own category instead of under weapons.  Added z as the default key.

https://cdn.discordapp.com/attachments/266345072554016768/586514092643188760/Controlkeyspanel.png

(these are the default controls)

Example in action: https://streamable.com/b3fk5

### CODE REVIEW

**Comment by Zitchas** [comment]
I don't have access to a system that can test this, so my question is thus: Does the autosteer act like a movement key, where the ship stops turning/tracking as soon as one lets go of the key? Or does it act like landing or boarding keys where one taps the key and it keeps being active until one taps something to break it?

I'm not sure which one I'd prefer, although I lean towards the former rather than the latter.

**Comment by ChamEV** [comment]
It acts like a movement key, it only works while you're holding it down.

File: source/PreferencesPanel.cpp [hunk 1]
```diff
@@ -332,7 +333,9 @@ void PreferencesPanel::DrawControls()
 		Command::FIGHT,
 		Command::GATHER,
 		Command::HOLD,
-		Command::AMMO
+		Command::AMMO,
+		Command::NONE,
```
Comment (jafdy) [inline source/PreferencesPanel.cpp]: This line appears extraneous (although I may be wrong).

File: source/AI.cpp [hunk 2]
```diff
@@ -1789,7 +1789,7 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 }
 
 
-
```
Comment (jafdy) [inline source/AI.cpp]: Whitespace changes.

**Comment by jafdy** [comment]
How is this different to the Automatic Aiming option, and how does it do a better job to justify the extra key it wants to use.

**Comment by Zitchas** [comment]
I think the main difference between this and the automatic aiming option is the fact that it is much more fine tuned: If it only works when the player is holding down the relevant button, so it is only working when called upon instead of being an "always on / always off" toggle.

I know I, personally, would like to be able to use an auto-aiming function some of the time, but I find it annoying most of the time. Since most of the time trumps sometimes, I leave the option off and just make do without. So I'd personally use this a fair bit.

File: source/PreferencesPanel.cpp (hunk 1 above)
Comment (ChamEV) [reply source/PreferencesPanel.cpp]: NONE tells the interface not to draw a box for the category line, it corresponds with the CATEGORIES array just above.

File: source/AI.cpp (hunk 2 above)
Comment (ChamEV) [reply source/AI.cpp]: The three remaining white space changes align the tabs with the current indentation of the surrounding text.  In the case shown above, there was an extra tab when there was no indentation above or below (between functions).

By default code:blocks was stripping white space at the ends of lines from everything and I had to go back and fix all that so it wouldn't be distracting.  I left the 3 specific white space changes in as they aligned the tab of each line to the current indent used by the rest of the file.  (Code blocks also defaulted to windows line endings on everything, but that was easier to fix).

**Comment by ChamEV** [comment]
> 
> 
> How is this different to the Automatic Aiming option, and how does it do a better job to justify the extra key it wants to use.

Automatic aiming only works while firing, forcing the player to turn towards a target at times when they may not want to as opposed to firing turrets and flying by.

In cases where there's a specific ship, perhaps disabled on the edge of the battle, you can select it and find which direction it's in easily.  If there's a ton of disabled other ships that you don't want (small fighters vs the big heavy cruiser that you want).

**Comment by warp-core** [comment]
@ChamEV are you still interested in getting this merged? Could you bring it up to date and resolve the conflicts? I'd like to test it, but it doesn't build.

File: source/AI.cpp [hunk 3]
```diff
@@ -3348,8 +3348,25 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player)
 		if(keyHeld.Has(AutopilotCancelKeys()))
 			keyStuck = keyHeld;
 	}
+	
 	bool shouldAutoAim = false;
-	if(Preferences::Has("Automatic aiming") && !command.Turn() && !ship.IsBoarding()
+	if( keyHeld.Has(Command::AUTOSTEER) && !command.Turn() && !ship.IsBoarding()
```
Comment (warp-core) [inline source/AI.cpp:3353]:
```suggestion
	if(keyHeld.Has(Command::AUTOSTEER) && !command.Turn() && !ship.IsBoarding()
```

**Comment by NickGrease** [comment]
I posted about this back in 2016 (#1258) which was a dup from 2015 (#166). I really hope this gets merged, it's been a long time coming.

**Comment by Amazinite** [comment]
Closing all PRs which have been marked "orphaned."

**Comment by UnorderedSigh** [comment]
I have resurrected this in #8508

**Pull request closed without merging**